libdnf.spec

```diff
@@ -28,11 +28,15 @@ Source0:        %{url}/archive/%{version}/libdnf-%{version}.tar.gz
 %bcond_without python3
 %bcond_without ruby
 
+%bcond_with    clang
 %bcond_with    sanitizers
 %bcond_without tests
 %bcond_with    performance_tests
 %bcond_with    dnfdaemon_tests
 
+%if %{with clang}
+    %global toolchain clang
+%endif
```

Comment:
How about adding the following down below?
```
%if %{with clang}
BuildRequires:  clang
%else
BuildRequires:  gcc-c++
%endif
```

Reply:
That would probably be required, yes... 😉

Reply:
Fixed, thanks, I also added a commit adding CI for building with clang, tested here: https://github.com/lukash/libdnf/actions/runs/729774009